### PR TITLE
Rename all instances of stop tactic to halt tactic

### LINKED ISSUE #2630: Rename StopTactic to HaltTactic
### Description of the task

StopTactic is too easily confused with the StopPlay, which doesn't involve halting all the robots. Renaming StopTactic to HaltTactic would be more consistent with the HaltPlay.

### Acceptance criteria

- [ ] Rename StopTactic to HaltTactic

### PR DESCRIPTION
**Pull request by vladakozachok**
### Resolved Issues

resolves #2630 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue

### CODE REVIEW

**Review by germchow — COMMENTED** [review]
I think it would be good to also rename the StopFSM (stop_fsm.h, stop_fsm.cpp, stop_fsm_test.cpp) to use the term halt instead of stop

Possibly also rename the stop directory (src/software/ai/hl/stp/tactic/stop) to halt

**Review by nimazareian — COMMENTED** [review]
The refactoring looks good to me! Just pending CI. Could you pull the upstream master branch into your PR branch? That should fix CI.

File: src/software/ai/hl/stp/play/ball_placement_play.cpp
Comment (itsarune) [inline src/software/ai/hl/stp/play/ball_placement_play.cpp]: huh it looks like we don't use this import in this file, so we can remove it

File: src/software/ai/hl/stp/play/corner_kick_play.cpp
Comment (itsarune) [inline src/software/ai/hl/stp/play/corner_kick_play.cpp]: it looks like we don't use this import, so it's safe to remove

File: src/software/ai/hl/stp/play/penalty_kick/penalty_kick_play_fsm.h
Comment (itsarune) [inline src/software/ai/hl/stp/play/penalty_kick/penalty_kick_play_fsm.h]: if it is unused, we can remove it

File: src/software/ai/hl/stp/tactic/halt/halt_tactic.h
Comment (itsarune) [inline src/software/ai/hl/stp/tactic/halt/halt_tactic.h]: the tactic string formatting is a bit strange here

**Review by itsarune — COMMENTED** [review]
nice work! should be good to go after a few minor changes

**Review by williamckha — DISMISSED** [review]
LGTM! Pull upstream master and CI should pass

**Review by itsarune — DISMISSED** [review]
LGTM!

**Review by williamckha — DISMISSED** [review]

**Review by itsarune — COMMENTED** [review]
I think you have a bad merge, your last commit undid the requested PR changes: https://github.com/UBC-Thunderbots/Software/pull/3037/commits/918acf18ac6d27347564ba94e0ff2868266a58ca

**Review by nimazareian — DISMISSED** [review]
Overall LGTM! :100: 

I noticed that some of @itsarune's comments are still open though.

File: src/software/ai/hl/stp/play/shoot_or_chip_play.cpp
Comment (itsarune) [inline src/software/ai/hl/stp/play/shoot_or_chip_play.cpp]: you can remove this import

File: src/software/ai/hl/stp/tactic/halt/halt_tactic.h
Comment (itsarune) [reply src/software/ai/hl/stp/tactic/halt/halt_tactic.h]: bump

File: src/software/ai/hl/stp/play/ball_placement_play.cpp
Comment (itsarune) [reply src/software/ai/hl/stp/play/ball_placement_play.cpp]: bump

File: src/software/ai/hl/stp/play/corner_kick_play.cpp
Comment (itsarune) [reply src/software/ai/hl/stp/play/corner_kick_play.cpp]: bump

File: src/software/ai/hl/stp/play/penalty_kick/penalty_kick_play_fsm.h
Comment (itsarune) [reply src/software/ai/hl/stp/play/penalty_kick/penalty_kick_play_fsm.h]: bump

**Review by itsarune — COMMENTED** [review]
there's some nits left + you need to pull upstream/master because there's a new github action

**Comment by nimazareian** [comment]
@vladakozachok Could you pull from master? There seems to be some conflicts. Also, I think Arun still has one outstanding comment.

**Review by williamckha — APPROVED** [review]

**Comment by nimazareian** [comment]
@vladakozachok Apologies for being slow in reviewing this PR, but there seems to be another conflict. Feel free to ping us here or on Discord once it's ready so we can try to review it before more conflicts.

**Review by itsarune — APPROVED** [review]
LGTM, pending merge confllct fix

**Comment by itsarune** [comment]
Stale, closing. Feel free to reopen once merged with master.

**Pull request closed without merging**
